pkg/controller/jobframework/webhook/defaulter.go

```diff
@@ -58,3 +70,32 @@ func (h *losslessDefaulter) Handle(ctx context.Context, req admission.Request) a
 	}
 	return response
 }
+
+// getUnknownPaths returns a list of unknown paths.
+func (h *losslessDefaulter) getUnknownPaths(raw []byte) (sets.Set[string], error) {
+	obj := h.object.DeepCopyObject()
+	strictErrors, err := json.UnmarshalStrict(raw, obj, json.DisallowUnknownFields)
+	if err != nil {
+		return nil, err
+	}
+	paths := sets.New[string]()
+	for _, err := range strictErrors {
+		paths.Insert(err.(json.FieldError).FieldPath())
```

Comment:
I'm wondering if it is safe to cast to `json.FieldError` unconditionally. Please check if other error types be present.

Still, even if this is the only error type possible currently it probably could change in the future, so I would suggest to check it first.

Reply:
Done